pkgs/applications/video/mkvtoolnix/default.nix

```diff
@@ -25,7 +25,7 @@ stdenv.mkDerivation rec {
     expat
     file xdg_utils boost libebml zlib
     libmatroska libogg libvorbis flac
-    (optional withGUI qt5.qtbase)
+    (optional withGUI qt5.qtbase.dev)
   ];
```

Comment:
> It means that optional withGUI qt5.qtbase (when withGUI is true) is a list with one element. 

Yes.

> This one element is itself a list with many elements.

No, it's just a different derivation. `qt5.qtbase` has multiple outputs: `out` and `dev`. It seems mkvtoolnix need a library in the `.dev` path.
I'm not sure how having a singleton list confuses mkDerivation.